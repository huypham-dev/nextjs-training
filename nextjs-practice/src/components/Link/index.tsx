import { memo } from "react";

import Link, { LinkProps } from "next/link";

interface Props extends LinkProps {
  children: React.ReactNode;
  className?: string;
}

const LinkBase = ({ children, className, ...linkProps }: Props) => (
  <Link {...linkProps}>
    <a className={className}>{children}</a>
  </Link>
);

export default memo(LinkBase);
