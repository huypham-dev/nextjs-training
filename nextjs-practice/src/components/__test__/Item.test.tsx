import { mount, shallow } from "enzyme";
import Link from "next/link";
import Image from "next/image";
import CategoryItem from "../Categories/Item";

describe("CategoryItem", () => {
  it("Snapshot", () => {
    const component = shallow(
      <CategoryItem imageUrl="/image.jpg" categoryName="mobile" />
    );

    expect(component).toMatchSnapshot();
  });

  it("Render component correctly with props", () => {
    const component = mount(
      <CategoryItem imageUrl="/image.jpg" categoryName="mobile" />
    );
    expect(component.containsMatchingElement(<p>MOBILE</p>)).toBeTruthy();
    expect(component.find(Link).length).toBe(1);
    expect(component.find(Link).prop("href")).toBe("/blog/mobile");
    expect(component.find(Image).prop("src")).toBe("/image.jpg");
  });
});
