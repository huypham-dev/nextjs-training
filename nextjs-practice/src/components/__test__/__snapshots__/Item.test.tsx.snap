// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem Snapshot 1`] = `
<div
  className="w-full md:w-1/2 lg:w-1/3 px-6 p-2"
>
  <Memo(LinkBase)
    className="block mb-8 text-center hover:opacity-70"
    href="/blog/mobile"
  >
    <Image
      alt="mobile"
      height={100}
      src="/image.jpg"
      width={100}
    />
  </Memo(LinkBase)>
  <Memo(Typography)
    className="text-xl text-center"
  >
    MOBILE
  </Memo(Typography)>
</div>
`;
